samples/chaincode/cc-tools-demo/README.md

```diff
@@ -0,0 +1,329 @@
+
+The `main.go` contains the starting point of the chaincode.
+
+To use FPC, we need to add some logic to instantiate our private chaincode and start it.
+To do so, we use `shim.Start(fpc.NewPrivateChaincode(new(CCDemo)))` and since we're already in the FPC repo we will point to the local FPC package.
+
+Go to the `go.mod` file in the `cc-tools-demo` chaincode folder and add the following replace line before the `require( )` block:
+
+```go
+replace github.com/hyperledger/fabric-private-chaincode => ../../../ 
+```
+
+CC-tools-demo chaincode has its own packages that are needed, so we run `go mod tidy` in the `cc-tools-demo` folder.
+
+Then, edit the chaincode in the `$FPC_PATH/samples/chaincode/cc-tools-demo/main.go` file:
+
+For import block, add the fpc package
+
+```go
+import (
+	///Keep everything as is
+	fpc "github.com/hyperledger/fabric-private-chaincode/ecc_go/chaincode"
+)
+```
+
+In the `main()` function, replace the following `if` statement:
+
+```go
+if os.Getenv("RUN_CCAAS") == "true" {
+	err = runCCaaS()
+} else {
+	err = shim.Start(new(CCDemo))
+}
+
+```
+
+With this:
+
+```go
+if os.Getenv("RUN_CCAAS") == "true" {
+	err = runCCaaS()
+} else {
+	if os.Getenv("FPC_ENABLED") == "true" {
+		err = shim.Start(fpc.NewPrivateChaincode(new(CCDemo)))
+	} else {
+		err = shim.Start(new(CCDemo))
+	}
+}
+```
+
+In the `runCCaaS()` function, replace the following line:
+
+```go
+ccid := os.Getenv("CHAINCODE_ID")
+```
+
+With this
+
+```go
+ccid := os.Getenv("CHAINCODE_PKG_ID")
+```
+
+And replace this
+
+```go
+server := &shim.ChaincodeServer{
+	CCID:     ccid,
+	Address:  address,
+	CC:       new(CCDemo),
+	TLSProps: *tlsProps,
+}
+```
+
+With this
+
+```go
+
+var cc shim.Chaincode
+if os.Getenv("FPC_ENABLED") == "true" {
+	cc = fpc.NewPrivateChaincode(new(CCDemo))
+} else {
+	cc = new(CCDemo)
+}
+
+server := &shim.ChaincodeServer{
+	CCID:     ccid,
+	Address:  address,
+	CC:       cc,
+	TLSProps: *tlsProps,
+}
+
+```
+
+## Building FPC Go Chaincode
+
+First, to update the go dependencies run (inside the dev environment):
+
+```bash
+cd $FPC_PATH/samples/chaincode/cc-tools-demo
+go get github.com/hyperledger/fabric-private-chaincode
+go mod tidy
+go get
+```
+
+Create a `Makefile` (i.e., `touch $FPC_PATH/samples/chaincode/cc-tools-demo/Makefile`) with the following content:
+
+```Makefile
+TOP = ../../..
+include $(TOP)/ecc_go/build.mk
+
+CC_NAME ?= fpc-cc-tools-demo
+
+EGO_CONFIG_FILE = $(FPC_PATH)/samples/chaincode/cc-tools-demo/ccToolsDemoEnclave.json
+ECC_MAIN_FILES=$(FPC_PATH)/samples/chaincode/cc-tools-demo
+
+```
+
+Please make sure that in the file above the variable `TOP` points to the FPC root directory (i.e., `$FPC_PATH`) as it uses the `$FPC_PATH/ecc_go/build.mk` file.
+
+In `$FPC_PATH/samples/chaincode/cc-tools-demo` directory, to build the chaincode and package it as docker image, execute:
+
+```bash
+make
```

Comment:
So I would add a note here that for those that have arm-based computers they should use the method recommended in the main readme for building both the cc-tools-demo chaincode as well as ercc